### ISSUE TITLE
[FR]: ssh_dispatch_run_fatal: Connection from ADDR port PORT: (message authentication code incorrect|Connection corrupted) [preauth]

### ISSUE THREAD

**Issue by OpenNya** [fail2ban/fail2ban#3486]
The [sshd] filter ignores the following two types of errors from the auth.log.

~~Both messages appear to indicate an attacker's attempt to exploit one or more of the known vulnerabilities in certain versions/builds of OpenSSH.
I believe that such events indicate a high risk of danger from the connecting hosts, so events must be detected by Fail2Ban via the "common failure" regex filter set.~~

### Environment:

- Fail2Ban version  : 0.11.1
- OS, including release name/version : Ubuntu 20.04.6 LTS

#### Service, project or product which log or journal should be monitored

- Name of filter or jail in Fail2Ban (if already exists) : sshd
- Service, project or product name, including release name/version : OpenSSH_8.2p1 Ubuntu-4ubuntu0.5, OpenSSL 1.1.1f

#### Log or journal information

- Log file name(s) : /var/log/auth.log

#### failures in sense of fail2ban filter (fail2ban must match):

```
Feb 26 02:58:43 vul-fra1 sshd[71905]: ssh_dispatch_run_fatal: Connection from 108.61.210.117 port 33738: message authentication code incorrect [preauth]
Feb 27 16:07:59 vul-fra1 sshd[80348]: ssh_dispatch_run_fatal: Connection from 108.61.210.117 port 52452: Connection corrupted [preauth]
```

**Comment by sebres** [comment]
> I believe that such events indicate a high risk of danger from the connecting hosts

Indeed it may be, but I wouldn't rule out messages like this by simple bad line or line abortion

> so events must be detected by Fail2Ban via the "common failure" regex filter set.

Well, partially agree... Just at the moment `normal` mode is intended to match only authentication failures and similar circumstances, so considering afore-mentioned too I'm still unsure we shouldn't put it to `extra` or `ddos` (and thus to `aggressive`) mode either, like 8bf15db688ac0f95f6f1489947ee00825340b23e (fix for #3169).

**Comment by sebres** [comment]
And again (the same as by #3485), are that messages alone in log?
If you see something else for that sessions better provide the whole excerpt, please.

**Comment by OpenNya** [comment]
> Indeed it may be, but I wouldn't rule out messages like this by simple bad line or line abortion

**Regarding sshd[71905]:**

There was one preceding message for that session.

```
Feb 26 02:58:43 vul-fra1 sshd[71905]: Bad packet length 1272247522. [preauth]
Feb 26 02:58:43 vul-fra1 sshd[71905]: ssh_dispatch_run_fatal: Connection from 44.44.44.44 port 33738: message authentication code incorrect [preauth] 
```
None of the messages above were caught by Fail2Ban.

I searched through all the system logs for this IP and found one UFW event related to this case.
There was a port scan performed earlier from the same host:

```
Mar 19 19:44:23 vul-fra1 kernel: [1014171.220571] [UFW BLOCK] IN=eth0 OUT= MAC=ea:08:8c:00:60:d0:9a:7e:8c:17:ff:f2:08:00 SRC=44.44.44.44 DST=11.11.11.11 LEN=40 TOS=0x00 PREC=0x00 TTL=51 ID=12426 PROTO=TCP SPT=2750 DPT=22222 WINDOW=34077 RES=0x00 SYN URGP=0  
```
From the first message "Bad packet length 1272247522" it becomes clear that we are dealing with a regular Buffer Overflow attack.
It looks like the attacker is trying to sent a huge amount of data in one request to exploit some buffer memory overflow vulnerability.

I think that such type of events have no relation with DDOS or AGGRESSIVE mode.
It would be logical to ban such kind of dangerous activity according to common (general) failure regexp.

**Regarding sshd[80348]:**

After searching by session ID and by IP i didn't find any additional related records.
The only side event I found was a port scan some time earlier, just like in the above.
```
Feb 27 16:07:59 vul-fra1 sshd[80348]: ssh_dispatch_run_fatal: Connection from 55.55.55.55 port 52452: Connection corrupted [preauth] 
```

**Comment by sebres** [comment]
> From the first message "Bad packet length 1272247522" it becomes clear that we are dealing with a regular Buffer Overflow attack.

Or it is simply a protocol mismatch (unexpected data in buffer), for instance from the port scanning tool, similar to #2404.

> I think that such type of events have no relation with DDOS or AGGRESSIVE mode.

Well to an extent it is rather matter of taste. At the moment those modes (and historically still one mode extra, which is also part of aggressive) are intended for everything excepting authentication issues or similar events (like user unknown, not allowed, locked, etc).

Anyway I'll add them to mode `ddos` (and therefore to `aggressive`).
Thanks!

**Comment by OpenNya** [comment]
> Or it is simply a protocol mismatch (unexpected data in buffer), for instance from the port scanning tool, similar to #2404.

After some googling I have found that these errors are not related to any Buffer Overflow vulnerability.
It could be buggy or outdated SSH client implementation, protocol mismatch, brute force etc.

The messages mentioned in the issue caption indicates that the connecting host is a port scanner and should be banned in the case of DDOS or AGGRESSIVE mode is used.

PS: I have edited my previous posts to remove any mention of danger or high risk.

**Comment by OpenNya** [comment]
> Anyway I'll add them to mode `ddos` (and therefore to `aggressive`).

Thanks for your help!
We'll wait for the commit ;)

**Comment by sebres** [comment]
Another candidate for ddos/aggressive mode:
```
2024-02-13T16:03:31.339080+01:00 srv sshd[12345]: fatal: Timeout before authentication for 192.0.2.123 port 55555
```
the messages like that seem to be mostly alone (no further disconnect etc).

to enhance:
```diff
  mdre-ddos = ...
              ...
+             ^Timeout before authentication for <ADDR>
```
or
```diff
- mdre-ddos = ^Did not receive identification string from <HOST>
+ mdre-ddos = ^(?:Did not receive identification string from|Timeout before authentication for) <ADDR>
```

**Comment by sebres** [comment]
implemented in 4f679a56e0d564bafa7781938f8d28bb69fd96af, CI passed, merge.
